Kudu.Core/Infrastructure/FileSystemHelpers.cs

```diff
@@ -130,6 +130,15 @@ public static DateTime GetLastWriteTimeUtc(string path)
             return Instance.File.GetLastWriteTimeUtc(path);
         }
 
+        // Get the latest LastWriteTime of all the files in a folder
+        public static DateTime GetFolderLatestWriteTimeUtc(string path)
+        {
+            DirectoryInfoBase jobBinariesDirectory = DirectoryInfoFromDirectoryName(path);
+            FileInfoBase[] files = jobBinariesDirectory.GetFiles("*.*", SearchOption.AllDirectories);
+
+            return files.Select(f => f.LastWriteTimeUtc).OrderByDescending(t => t).FirstOrDefault();
```

Comment:
is it possible for a file with a LastWriteTime that is not the latest one in the folder to be updated (and still have its LastWriteTime as not the latest)?
Also LastWriteTime could be sometime in the future (for whatever reason) which can be a problem.